### PR TITLE
Feature: Publish the library

### PR DESCRIPTION
**Pull request by kristinpeterson**
**Goal**: publish library to [Maven Central Repository](http://search.maven.org/)

**Pre-requisites** (outlined in [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html)):
- [x] [Create a ticket with Sonatype](http://central.sonatype.org/pages/ossrh-guide.html#create-a-ticket-with-sonatype) which uses JIRA to manage requests:
 - [x] Create JIRA account
 - [x] Create a New Project ticket
- [x] Review [requirements](http://central.sonatype.org/pages/requirements.html) 
- [x] Setup [PGP](http://central.sonatype.org/pages/working-with-pgp-signatures.html) signatures
- [x] [Deployment](http://central.sonatype.org/pages/ossrh-guide.html#deployment)
 - [x] Use [Gradle](http://central.sonatype.org/pages/gradle.html) to automate deployment process
- [x] [Releasing to Central](http://central.sonatype.org/pages/ossrh-guide.html#releasing-to-central)

Release pending approval from Maven Central Repository (one-time approval required for first release)

### CODE REVIEW

**Comment by kristinpeterson** [comment]
@fsaravia ready for review!

Updates to `build.gradle` per [central repository guidelines](http://central.sonatype.org/pages/gradle.html), save for minor edits to `sourcesJar` & `javadocJar` tasks which required some tweaking because this is an Android library and the instructions provided are for Java libraries.

Note: the added `uploadArchives` task runs successfully, and works as expected: uploads all necesary components to Central Repository.  The library is now in the inital review stages w/ Central Repo.

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,8 @@
 .gradle
 /local.properties
-/.idea/workspace.xml
-/.idea/libraries
+.idea
 .DS_Store
 /build
 app/app.iml
-m2x-android.iml
```
Comment (fsaravia) [inline .gitignore]: @kristinpeterson do you mind configuring your editor to add a blank line at the end of each file please?

File: app/build.gradle [hunk 2]
```diff
@@ -24,3 +29,75 @@ dependencies {
...
+                name 'AT&T M2X Android Library'
+                packaging 'jar'
+                artifactId 'android'
+                description 'An Android library for interacting with the AT&amp;T M2X API.'
+                url 'http://m2x.att.com'
+
+                scm {
+                    connection 'scm:git:git@github.com:attm2x/m2x-android.git'
+                    developerConnection 'scm:git:git@github.com:attm2x/m2x-android.git'
+                    url 'git@github.com:attm2x/m2x-android.git'
```
Comment (fsaravia) [inline app/build.gradle]: If I'm not mistaken, you need to change these `git@github.com` urls to their `https` versions which is the one non-contributors use to access the repository

**Comment by fsaravia** [comment]
@kristinpeterson other than those comments, this looks great. Thanks!

File: .gitignore (hunk 1 above)
Comment (kristinpeterson) [inline .gitignore]: Don't mind at all! Thanks for catching this!

File: app/build.gradle (hunk 2 above)
Comment (kristinpeterson) [inline app/build.gradle]: I formatted the `scm` block according to [this section](http://central.sonatype.org/pages/requirements.html#scm-information) of the requirements docs.

Though I notice that all other scm's in that section of the docs use `http`/`https`.  I've contacted their support team to obtain clarification.  Will update here once they respond.

Comment (kristinpeterson) [inline app/build.gradle]: Well that was fast...response from their support team:

> It's fine to use those style URLs in the POM element

Comment (fsaravia) [inline app/build.gradle]: Great!

**Comment by kristinpeterson** [comment]
@fsaravia thanks!

**Pull request merged**